### PR TITLE
fix(src/handler.ts): drop role with permissions

### LINKED ISSUE #22: Role creation sometimes errors with Tuple concurrently updated
Role creation sometimes fails with the `Tuple concurrently updated`. A retry mechanism might be sufficient.

```
2024-06-11T20:51:53.067Z	f21a5213-3ca7-469d-aef9-43cbd7f60bda	DEBUG	Executing SQL [
  'start transaction',
  "create role anondata_rw with login password '<the password>'",
  'DO $$BEGIN\n' +
    `IF EXISTS (select from pg_database where datname = 'anonymizedData' and datistemplate = false) THEN grant connect on database "anonymizedData" to anondata_rw; END IF;\n` +
    'END$$;',
  'commit'
]
```

```
2024-06-11T20:51:53.067Z f21a5213-3ca7-469d-aef9-43cbd7f60bda DEBUG Executing SQL [ 'start transaction', "create role anondata_rw with login password '<the password>'", 'DO $$BEGIN\n' + `IF EXISTS (select from pg_database where datname = 'anonymizedData' and datistemplate = false) THEN grant connect on database "anonymizedData" to anondata_rw; END IF;\n` + 'END$$;', 'commit' ]
2024-06-11T20:51:55.928Z
2024-06-11T20:51:55.928Z	f21a5213-3ca7-469d-aef9-43cbd7f60bda	ERROR	Invoke Error 	
{
    "errorType": "error",
    "errorMessage": "tuple concurrently updated",
    "code": "XX000",
    "length": 225,
    "name": "error",
    "severity": "ERROR",
    "where": "SQL statement \"grant connect on database \"anonymizedData\" to anondata_rw\"\nPL/pgSQL function inline_code_block line 2 at SQL statement",
    "file": "heapam.c",
    "line": "4665",
    "routine": "simple_heap_update",
    "stack": [
        "error: tuple concurrently updated",
        "    at <anonymous> (/node_modules/cdk-rds-sql/node_modules/pg/lib/client.js:526:17)",
        "    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)",
        "    at <anonymous> (/node_modules/cdk-rds-sql/src/handler.ts:313:13)",
        "    at async Promise.all (index 2)",
        "    at Runtime.exports.handler (/node_modules/cdk-rds-sql/src/handler.ts:311:9)"
    ]
}
```

### PR DESCRIPTION
**Pull request by ww-daniel-mora**
* Added logging to failed sql commands
* Allowed retry for concurrent delete
* Increased retry for concurrency error

### CODE REVIEW

**Comment by ww-daniel-mora** [comment]
example concurrent delete error
```
2024-06-12 17:52:38 UTC:10.255.6.210(19770):postgres@anonymizedData:[2425]:ERROR:  tuple concurrently deleted
```

File: src/handler.ts [hunk 1]
```diff
@@ -142,15 +154,19 @@ const jumpTable: JumpTable = {
...
-      // TODO: if user is owner of a database, assign ownership to master user?
+    Delete: (resourceId: string, _props: RoleProps) => {
       return [
         "start transaction",
         format(
-          "DO $$BEGIN\nIF EXISTS (select from pg_catalog.pg_roles WHERE rolname = '%s') AND EXISTS (select from pg_database WHERE datname = '%s') THEN revoke all privileges on database %I from %I; END IF;\nEND$$;",
+          `DO $$
+BEGIN
+  IF EXISTS (select from pg_catalog.pg_roles WHERE rolname = '%s') THEN
+    reassign OWNED by %I to posgres;
```
Comment (berenddeboer) [inline src/handler.ts:164]: You have a hard-coded role here, which people may not have. Also from that discussion reassigning can lead to privilege escalation, so not keen on that approach. Can you remove this from the PR? I can merge most of it, but not this bit.

Comment (ww-daniel-mora) [reply src/handler.ts:164]: Will do.

Comment (berenddeboer) [reply src/handler.ts:164]: From [that thread](https://stackoverflow.com/posts/54527768/revisions), I suggest you add something like this:

```sql
DO $$ 
DECLARE
    r RECORD;
BEGIN
    FOR r IN (SELECT schemaname FROM pg_schemas) LOOP
      EXECUTE 'REVOKE ALL PRIVILEGES ON ALL TABLES IN SCHEMA ' || quote_ident(r.schemaname) || ' FROM %s';
      EXECUTE 'REVOKE ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA ' || quote_ident(r.schemaname) || ' FROM %s';
      EXECUTE 'REVOKE ALL PRIVILEGES ON ALL FUNCTIONS IN SCHEMA ' || quote_ident(r.schemaname) || ' FROM %s';
      EXECUTE 'REVOKE ALL ON SCHEMA ' || quote_ident(r.schemaname) || ' FROM %I';
    END LOOP;
END $$;
```

Typing this from memory, and needs testing.

And you could add:

```sql
REVOKE CONNECT ON DATABASE db_name FROM role_name;
```

Keep the drop OWNED by %I as the first statement.

Comment (berenddeboer) [reply src/handler.ts:164]: But this all needs a bit of work, I prefer it if you move this to a separate PR.

**Comment by ww-daniel-mora** [comment]
Ok for now I have reverted to the existing deletion behavior.

This change now
* Adds logging when retrying
* Expands retry filter to include concurrent delete
* Doubles the max retry count

File: src/handler.ts (hunk 1 above)
Comment (ww-daniel-mora) [reply src/handler.ts]: Sequences specifically is what cause my failure but I can work that into a seperate PR

**Pull request merged**